src/containers/AdminCampaignEdit.jsx

```diff
@@ -202,17 +206,37 @@ export class AdminCampaignEdit extends React.Component {
     // 3. Refetch/poll updates data in loadData component wrapper
     //    and triggers *this* method => this.props.campaignData => this.state.campaignFormValues
     // So campaignFormValues should always be the diffs between server and client form data
-    let { expandedSection, isPolling } = this.state;
+    let { expandedSection } = this.state;
     let expandedKeys = [];
     if (expandedSection !== null) {
       expandedSection = this.sections()[expandedSection];
       expandedKeys = expandedSection.keys;
     }
 
+    const keysInUnsavedSections = [];
+    this.sections().forEach(section => {
+      if (!this.checkSectionSaved(section)) {
+        keysInUnsavedSections.push(...section.keys);
+      }
+    });
+
+    const newCampaignData = newProps.campaignData.campaign;
+    const changedKeysInUnsavedSections = [];
+    keysInUnsavedSections.forEach(key => {
+      if (newCampaignData[key] !== this.state.campaignFormValues[key]) {
```

Comment:
Do we need a deep equality comparison here?  Didn't new data come in from the server?  Since we already have lodash in the codebase, maybe use: `_.isEqual()`.
Regardless, let's add a comment for line 226 to explain what we are and are not updating/testing.

Reply:
@schuyler1d No, we're going key by key to find keys that were changed by the user. If a key was changed by the user we remove it from the update from the server.

Reply:
ok, so this was changed by the user, not by the server.  Should we also see if the original version is different than the new server version to see if the user might be overwriting a change? (Merging is probably WAY too complex, etc, but we might at least want to warn folks)